cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -375,7 +380,12 @@ private void restartIfNecessary(NodeRef nodeRef, RestartContext restartContext)
             LOGGER.debugCr(reconciliation, "Waiting for pod {} to become ready before checking its state", nodeRef.podName());
             try {
                 await(isReady(pod), operationTimeoutMs, TimeUnit.MILLISECONDS, e -> new RuntimeException(e));
-            } catch (Exception e)   {
+            } catch (Exception e) {
+                BrokerState brokerState = kafkaAgentClient.getBrokerState(pod.getMetadata().getName());
```

Comment:
If the `kafkaAgentClient` is used only here and if the `kafkaAgentClient` is not initialized in the constructor -> would it make sense to initialize it only here?

Reply:
yes I agree with this. I will move it to here.